[Banner] Fix top layout constraint in Autolayout example on pre iOS 11 OSes.

fluxminds opened the pull request:
closes https://github.com/material-components/material-components-ios/issues/8824.

Previously, the banner under autolayout constraints relied on `layoutMarginsGuide` which takes `addtionSafeAreaInsets` from flexibleHeader for iOS 11+. This fix switches it back to using `topLayoutGuide` to make it compatible with iOS 7+ when iOS11 is not avaliable.

| Screenshot on iPhone7 10.2 after the fix |
| --- |
| ![Simulator Screen Shot - iPhone 7 - 2019-12-02 at 16 06 01](https://user-images.githubusercontent.com/8836258/69995422-08db1880-151e-11ea-8ee9-23e580cccd79.png) |

Commit:
[Banner] Fix top layout constraint in Autolayout example for pre iOS 11 OSes.

material-automation commented:
bazel detected changes to the following targets:

```
//components/Banner:ObjcExamples
```

<!-- identifier: affected-targets -->

bryanoltman commented:
Does this change affect behavior on iOS 11 and above?

Commit:
CRC changes.

fluxminds commented:
> Does this change affect behavior on iOS 11 and above?

Addressed in 31b0070554922e82361b7cc74f88f5bb58def800. It doesn't, but since `topLayoutGuide` is deprecated in iOS11 above, I changed the usage.

Commit:
format

material-automation commented on components/Banner/examples/BannerAutolayoutExampleViewController.m:
```diff
@@ -67,8 +67,13 @@ - (void)viewDidLoad {
-      [bannerView.topAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.topAnchor];
+  NSLayoutConstraint *bannerViewConstraintTop;
+  if (@available(iOS 11.0, *)) {
+    bannerViewConstraintTop = [bannerView.topAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.topAnchor];
```
clang-format suggested the following change:


```
-    bannerViewConstraintTop = [bannerView.topAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.topAnchor];
+    bannerViewConstraintTop =
+        [bannerView.topAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.topAnchor];
```

<!-- identifier: clang-format -->
<!-- File: components/Banner/examples/BannerAutolayoutExampleViewController.m -->

material-automation commented on components/Banner/examples/BannerAutolayoutExampleViewController.m:
```diff
@@ -67,8 +67,13 @@ - (void)viewDidLoad {
+    bannerViewConstraintTop = [bannerView.topAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.topAnchor];
+  } else {
+    bannerViewConstraintTop =
+    [bannerView.topAnchor constraintEqualToAnchor:self.topLayoutGuide.bottomAnchor];
```
clang-format suggested the following change:


```
-    [bannerView.topAnchor constraintEqualToAnchor:self.topLayoutGuide.bottomAnchor];
+        [bannerView.topAnchor constraintEqualToAnchor:self.topLayoutGuide.bottomAnchor];
```

<!-- identifier: clang-format -->
<!-- File: components/Banner/examples/BannerAutolayoutExampleViewController.m -->

material-automation commented:
clang-format recommended changes.

Consider installing [git-clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) and running the following command:

```bash
git clang-format $(git merge-base origin/develop HEAD)
```

<!-- identifier: clang-format -->

codeman7 reviewed (commented):
~Can we use `topLayoutGuide` for both and remove the if statement altogether or is this required to play nicely with the FlexibleHeader?~ Please ignore I went to fast and didn't see the existing thread.
